pkg/dochandler/handler.go

```diff
@@ -157,30 +165,44 @@ func (r *DocumentHandler) getCreateResponse(op *operation.Operation, pv protocol
 		return nil, err
 	}
 
+	ti := r.getTransformationInfoForUnpublished(op.UniqueSuffix, "")
+
+	return pv.DocumentTransformer().TransformDocument(rm, ti)
+}
+
+func (r *DocumentHandler) getTransformationInfoForUnpublished(suffix string, createRequestJCS string) protocol.TransformationInfo {
 	ti := make(protocol.TransformationInfo)
 	ti[document.PublishedProperty] = false
 
-	id := r.namespace + docutil.NamespaceDelimiter + op.UniqueSuffix
+	id := fmt.Sprintf("%s:%s", r.namespace, suffix)
+
+	// For interim/unpublished documents we should set optional label if specified.
+	if r.label != "" {
+		id = fmt.Sprintf("%s:%s:%s", r.namespace, r.label, suffix)
+	}
 
-	// if optional domain is specified we should set equivalent id with domain hint for interim/unpublished documents
-	if r.domain != "" {
-		id = r.namespace + docutil.NamespaceDelimiter + "interim" + docutil.NamespaceDelimiter + op.UniqueSuffix
+	var equivalentIDs []string
 
-		equivalentID := r.namespace + docutil.NamespaceDelimiter +
-			"interim" + docutil.NamespaceDelimiter + r.domain + docutil.NamespaceDelimiter + op.UniqueSuffix
+	if createRequestJCS != "" {
+		// we should always set short form equivalent id for long form resolution
+		equivalentIDs = append(equivalentIDs, id)
+	}
 
-		ti[document.EquivalentIDProperty] = []string{equivalentID}
+	// Also, if optional domain is specified, we should set equivalent id with domain hint
+	if r.label != "" && r.domain != "" {
+		equivalentID := fmt.Sprintf("%s:%s:%s:%s", r.namespace, r.label, r.domain, suffix)
```

Comment:
what about docutil.NamespaceDelimiter

Reply:
I find it easier to follow with Sprintf and it is no longer just namespace delimiter but just delimiter for different parts of did